core/src/main/java/org/elasticsearch/ingest/ConfigurationUtils.java

```diff
@@ -251,6 +250,28 @@ public static ElasticsearchParseException newConfigurationException(String proce
         return processors;
     }
 
+    public static TemplateService.Template compileTemplate(String processorType, String processorTag, String propertyName,
+                                                           String propertyValue, TemplateService templateService) {
+        try {
+            return templateService.compile(propertyValue);
+        } catch (Exception e) {
```

Comment:
we've been using `ElasticsearchParseException` as the exception container for all our pipeline build-time configuration exceptions.

> it means bugs in compilation get converted into a parse exception that sounds like the user can correct

why would the user not be able to correct these exceptions?